controllers/coordination/integration/integration_suite_test.go

```diff
@@ -22,22 +23,31 @@ func TestIntegration(t *testing.T) {
 }
 
 var (
-	testEnv   *envtest.Environment
-	k8sClient client.Client
+	testEnv           *envtest.Environment
+	k8sClient         client.Client
+	controllersClient client.Client
 )
 
 var _ = BeforeSuite(func() {
 	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
 
 	testEnv = &envtest.Environment{}
 
-	cfg, err := testEnv.Start()
+	adminConfig, err := testEnv.Start()
 	Expect(err).NotTo(HaveOccurred())
-	Expect(cfg).NotTo(BeNil())
 
-	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	k8sClient, err = client.New(adminConfig, client.Options{Scheme: scheme.Scheme})
+	Expect(err).NotTo(HaveOccurred())
+
+	controllersUser, err := testEnv.ControlPlane.AddUser(envtest.User{Name: "envtest-controller"}, adminConfig)
+	Expect(err).NotTo(HaveOccurred())
+	helpers.BindUserToControllersRole(k8sClient, "envtest-controller")
+
+	controllersClient, err = client.New(
```

Comment:
I think this client won't use a cache.

Reply:
I do not think cache is required for the name registry integration test. Or am I wrong?

Reply:
yeah, that is why we allowed get on leases as well: https://github.com/cloudfoundry/korifi/pull/2655/files#diff-90cb3e4b36d2d633be58cb91c3d2858b713bf7cb42e1f842b791936c4ce43ff4R30

Reply:
I've run this suite without the list and watch permissions on lease - it still passes. So I really think we need to activate the cache on the client here.